### ISSUE TITLE
Improvement, colors

### ISSUE THREAD

**Issue by Samsy** [Jam3/three-bmfont-text#14]
What about passing colors through geometry attributes instead as an uniform ?
So multiple text could share the same material / textures

**Comment by mattdesl** [comment]
Yeah, I think it would work though I'm curious how the API would look. Using attribute strings (where a paragraph is made up of chunks of strings with things like font style/color/size/etc) can get really complicated really quickly.

Ideally this module could handle all that stuff but it can become very application-dependent. Some apps might want to pack multiple font styles into the same atlas, so a chunk of text can be drawn in the same call. Other apps might prefer to implement color as a palette index to save vertex data (see [here](https://github.com/Jam3/three-bmfont-text/pull/7)).

Maybe another approach is to expose the glyphs in a nice way so that it's easy to build your own vertex attributes on top of them, suited to your application.
